js/editor-ta.js

```diff
@@ -67,10 +64,30 @@ function EditorTextArea(editorElement, settings) {
   this.element_.appendChild(this.mirror_);
   this.element_.appendChild(this.lineMirror_);
 
+  // The main text area the user will be interacting with.
+  this.attachTextArea(document.createElement('textarea'));
+
   this.calibrateDimensions();
 
   // TODO: set up search
-  // TODO: setup how we are handling tabs?
+  // TODO: setup how we are handling tab characters?
+}
+
+EditorTextArea.prototype.attachTextArea = function(textarea) {
+  const initFontSize = this.settings_.get('fontsize') + 'px';
+  // detach previous text area
```

Comment:
why is it that we need to detach the old and create a new? why can't we reuse the old?

Reply:
By passing around the textarea HTML element we preserve the undo/redo state and also avoid having to copy the whole test file over when we switch sessions.

Reply:
Ack